src/aof.c

```diff
@@ -1005,6 +1005,121 @@ int startAppendOnly(void) {
+    new_incr_filepath = makePath(server.aof_dirname, new_incr_filename);
+    newfd = open(new_incr_filepath, O_WRONLY | O_TRUNC | O_CREAT, 0644);
+    sdsfree(new_incr_filepath);
+    if (newfd == -1) {
+        serverLog(LL_WARNING, "Can't open the append-only file %s: %s", new_incr_filename, strerror(errno));
+        goto cleanup;
+    }
+
+    /* Change the AOF file type in 'incr_aof_list' from AOF_FILE_TYPE_INCR
+     * to AOF_FILE_TYPE_HIST, and move them to the 'history_aof_list'. */
+    markRewrittenIncrAofAsHistory(temp_am);
+
+    /* Persist AOF Manifest. */
+    if (persistAofManifest(temp_am) == C_ERR) {
+        goto cleanup;
+    }
+
+    /* Now, we can safely Update the server's manifest with our modified one */
+    aofManifestFreeAndUpdate(temp_am);
+
+    aofDelHistoryFiles();
+
+    sdsfree(new_base_filepath);
+
+    /* Set the initial repl_offset, which will be applied to fsynced_reploff */
+    atomic_store_explicit(&server.fsynced_reploff_pending, server.primary_repl_offset, memory_order_relaxed);
+    /* Update the fsynced replication offset that just now become valid. */
+    long long fsynced_reploff_pending =
+        atomic_load_explicit(&server.fsynced_reploff_pending, memory_order_relaxed);
+    server.fsynced_reploff = fsynced_reploff_pending;
+
+    /* If AOF fsync error in bio job, we just ignore it and log the event. */
+    int aof_bio_fsync_status = atomic_load_explicit(&server.aof_bio_fsync_status, memory_order_relaxed);
+    if (aof_bio_fsync_status == C_ERR) {
+        serverLog(LL_WARNING, "AOF reopen, just ignore the AOF fsync error in bio job");
+        atomic_store_explicit(&server.aof_bio_fsync_status, C_OK, memory_order_relaxed);
+    }
+
+    /* If AOF was in error state, we just ignore it and log the event. */
+    if (server.aof_last_write_status == C_ERR) {
+        serverLog(LL_WARNING, "AOF reopen, just ignore the last error.");
+        server.aof_last_write_status = C_OK;
+    }
+
+    /* Pretend the bgrewrite has been done successfully */
+    server.aof_lastbgrewrite_status = C_OK;
+    server.stat_aofrw_consecutive_failures = 0;
+
+    server.aof_last_fsync = server.mstime;
+    server.aof_last_incr_size = 0;
+    server.aof_last_incr_fsync_offset = 0;
+    server.aof_rewrite_base_size = getAppendOnlyFileSize(new_base_filename, NULL);
+    server.aof_current_size = server.aof_rewrite_base_size;
+
+    server.aof_fd = newfd;
+    server.aof_state = AOF_ON;
+
+    return C_OK;
+
+cleanup:
+    if (server.rdb_del_sync_files && allPersistenceDisabled()) {
+        serverLog(LL_NOTICE, "Removing the RDB file obtained from "
+                             "the primary. This replica has persistence "
+                             "disabled");
+        if (rdbfile_renamed) {
+            bg_unlink(new_base_filepath);
+        } else {
+            bg_unlink(server.rdb_filename);
+        }
+    }
+    if (temp_am) aofManifestFree(temp_am);
+    if (new_base_filename) sdsfree(new_base_filename);
```

Comment:
the 2 file names are just pointers of manifest, don't need to be freed.

and free the 2 file paths after where they are used would be better.